Mathlib/CategoryTheory/Products/Bifunctor.lean

```diff
@@ -56,3 +58,56 @@ theorem diagonal' (F : C × D ⥤ E) (X X' : C) (f : X ⟶ X') (Y Y' : D) (g : Y
 #align category_theory.bifunctor.diagonal' CategoryTheory.Bifunctor.diagonal'
 
 end CategoryTheory.Bifunctor
+
+namespace CategoryTheory.Functor
+
+/-- Opposite of a bifunctor.
+-/
+@[simps!]
+def biop (F : Cᵒᵖ ⥤ D ⥤ E) : C ⥤ Dᵒᵖ ⥤ Eᵒᵖ := F.rightOp ⋙ Functor.opHom _ _
+
+end CategoryTheory.Functor
+
+namespace CategoryTheory
+
+variable {C₁ C₂ : C} {D₁ D₂ : D} {E₁ E₂ : E}
+
+/-- Action of two-variable functors on objects. -/
+abbrev Functor.obj₂ (H : C ⥤ D ⥤ E) (A : C) (B : D) : E := (H.obj A).obj B
+
+/-- Action of three-variable functors on objects. -/
+abbrev Functor.obj₃ (H : C ⥤ D ⥤ E ⥤ F) (A : C) (B : D) (C : E) : F :=
+  ((H.obj A).obj B).obj C
+
+/-- Apply a natural transformation between bifunctors to two objects. -/
+abbrev NatTrans.app₂ {F G : C ⥤ D ⥤ E} (α : NatTrans F G) (X : C) (Y : D) :
+    F.obj₂ X Y ⟶ G.obj₂ X Y :=
+  (α.app X).app Y
+
+/-- Apply a natural transformation between bifunctors in three variables to three objects. -/
+abbrev NatTrans.app₃ {H G : C ⥤ D ⥤ E ⥤ F} (α : NatTrans H G) (X : C) (Y : D) (Z : E) :
+    H.obj₃ X Y Z ⟶ G.obj₃ X Y Z :=
+  ((α.app X).app Y).app Z
+
+/- Natural transformations between functors with many variables. -/
+namespace NatTrans
+
+@[reassoc]
+lemma comp_app₂ {H G K : C ⥤ D ⥤ E} (α : H ⟶ G) (β : G ⟶ K) (X : C) (Y : D) :
+    (α ≫ β).app₂ X Y = α.app₂ X Y ≫ β.app₂ X Y :=
+  by rw [app₂, comp_app, comp_app]
```

Comment:
Untested, but it should a `rfl` lemma, so that users may do `dsimp [comp_app₂]`
```suggestion
    (α ≫ β).app₂ X Y = α.app₂ X Y ≫ β.app₂ X Y := rfl
```
(Untested: this one may be a good `simp` lemma.)

Reply:
(Same for `comp_app₃`.)

Reply:
Good points, I've tagged them as `simp`. Using `@[reassoc, simp]` over `@[reassoc (attr := simp)]` is intentional, the linter complains about the latter

Reply:
Yes, `@[simp, reassoc]` is standard when compositions appear only in the RHS.